key/src/main/java/software/amazon/kms/key/UpdateHandler.java

```diff
@@ -1,77 +1,71 @@
-                        .done((describeKeyRequest, describeKeyResponse, proxyInvocation, model, context) -> {
+                        .done(describeKeyResponse -> {
                             resourceStateCheck(describeKeyResponse.keyMetadata());
 
-                            if (context.isKeyStatusRotationUpdated()) return ProgressEvent.progress(model, context);
-
-
-                            final boolean prevIsEnabled = previousModel.getEnabled();
-                            final boolean currIsEnabled = model.getEnabled();
-
-                            final boolean prevIsRotationEnabled = previousModel.getEnableKeyRotation();
-                            final boolean currIsRotationEnabled = model.getEnableKeyRotation();
-
-                            final boolean hasUpdatedStatus = prevIsEnabled ^ currIsEnabled;
-                            final boolean hasUpdatedRotation = prevIsRotationEnabled ^ currIsRotationEnabled;
-
-                            // if key is not being updated and is disabled then we cannot perform any updates on the key
-                            if (!prevIsEnabled && !hasUpdatedStatus && hasUpdatedRotation) // key stays disabled
-                                throw new CfnInvalidRequestException("You cannot modify the EnableKeyRotation property when the Enabled property is false. Set Enabled to true to modify the EnableKeyRotation property.");
-
-                            // if key is disabled then it needs to get enabled first and then update rotation if necessary
-                            // check if key has been enabled and propagated otherwise eventual inconsistency might occur and rotation status update might hit invalid state exception
-                            if (!prevIsEnabled && currIsEnabled && !context.isKeyEnabled())  // enable key
-                                return updateKeyStatus(proxy, proxyInvocation, model, context, true);
-
-                            // update rotation if necessary
-                            if (hasUpdatedRotation)
-                                updateKeyRotationStatus(proxy, proxyInvocation, model, context, currIsRotationEnabled);
-
-                            // disable the key if necessary
-                            // changing status from enabled to disabled wont affect other updates since they are possible with disabled key
-                            // rotation update happens before disabling key
-                            if (prevIsEnabled && !currIsEnabled) // disable key
-                                updateKeyStatus(proxy, proxyInvocation, model, context, false);
-
-                            context.setKeyStatusRotationUpdated(true);
-                            return ProgressEvent.progress(model, context);
+                            return progress;
                         })
                 )
+                .then(progress -> validateResourceModel(progress, previousModel, model))
```

Comment:
since it is used only here, maybe we can use it here as inline? basehandler already a bit overloaded

Reply:
This is also used by the CreateHandler